app/src/main/java/com/example/sdp_bfit/workout/WorkoutFragment.java

```diff
@@ -69,6 +89,40 @@ public void onChanged(@Nullable String s) {
         SC = root.findViewById(R.id.SC);
         SC.setText(stepCount.toString());
 
+        Savew = root.findViewById(R.id.Savew);
+
+        Savew.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+
+                int step = Integer.parseInt(SC.getText().toString());
+                Double dis=Double.valueOf(textView10.getText().toString());
+                Double cal=Double.valueOf(textView3.getText().toString());
+
+                Workout workout = new Workout( step, dis, cal, todayDate);
+                Database db = new Database(getContext());
+                boolean success = db.insertWorkout(workout);
+                if (success = true) {
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [QBA: Method assigns boolean literal in boolean expression (QBA_QUESTIONABLE_BOOLEAN_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#qba-method-assigns-boolean-literal-in-boolean-expression-qba-questionable-boolean-assignment).
This method assigns a literal boolean value (true or false) to a boolean variable inside an if or while expression. Most probably this was supposed to be a boolean comparison using ==, not an assignment using =.